packages/shared/src/hooks/utils/useGetShortUrl.ts

```diff
@@ -0,0 +1,45 @@
+import { useQueryClient } from '@tanstack/react-query';
+import request from 'graphql-request';
+import { useCallback } from 'react';
+import { graphqlUrl } from '../../lib/config';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { GET_SHORT_URL_QUERY } from '../../graphql/urlShortener';
+import { addTrackingQueryParams } from '../../lib/share';
+
+interface UseGetShortUrlResult {
+  getShortUrl: (url: string, cid?: string) => Promise<string>;
+}
+
+export const useGetShortUrl = (): UseGetShortUrlResult => {
+  const { user, isAuthReady } = useAuthContext();
+  const queryClient = useQueryClient();
+
+  const getShortUrl = useCallback(
+    async (url: string, cid?: string) => {
+      if (!url || !isAuthReady || !user) {
+        return url;
+      }
+
+      const trackingUrl = cid
+        ? addTrackingQueryParams({ link: url, userId: user?.id, cid })
+        : url;
+      const shortUrlKey = ['short_url', trackingUrl];
```

Comment:
For query keys you should use `generateQueryKey` it's a stronger typed object and does handling for you.